WC_Layers/OpenMC_PhotonTransport.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Aug  7 10:07:33 2024
+
+@author: Anupama Rajendra
+"""
+
+import openmc
+import argparse
+import numpy as np
+from Source_Mesh_Reader import summed_strengths, strengths_list
+from TwoLayers_Materials import *
+from TwoLayers_Geometry import *
+
+Mesh_File = 'OpenMC_Mesh.h5m'
+
+parser = argparse.ArgumentParser(description="Specify required inputs: file path to ALARA Element Library, element name, inner radius [cm], outer_radius [cm]")
+
+#Required user inputs:
+parser.add_argument('--filepath', type=str, required=True)    
+parser.add_argument('--element_1', type=str, required=True)
+parser.add_argument('--element_2', type=str, required=True)
+#Shell radii are left as user inputs, but the mesh is specific to W_inner_radius = 1000, W_outer_radius = 1005, C_inner_radius = 995
+parser.add_argument('--W_inner_radius', type=float, required=True)
+parser.add_argument('--W_outer_radius', type=float, required=True)
+parser.add_argument('--C_inner_radius', type=float, required=True)
+
+args = parser.parse_args()
+
+fp = args.filepath
+E_1 = args.element_1
+E_2 = args.element_2
+R_W_1 = args.W_inner_radius
+R_W_2 = args.W_outer_radius
+R_C_1 = args.C_inner_radius
+
+# fp = '../../ALARA/data/elelib.std'
+# E_1 = 'W'
+# E_2 = 'C'
+# R_W_1 = 1000
+# R_W_2 = 1005
+# R_C_1 = 995
+bounds = [0, 
+                 1.00e+4, 
+                 2.00e+4, 
+                 5.00e+4, 
+                 1.00e+5, 
+                 2.00e+5, 
+                 3.00e+5, 
+                 4.00e+5, 
+                 6.00e+5, 
+                 8.00e+5, 
+                 1.00e+6, 
+                 1.22e+6, 
+                 1.44e+6, 
+                 1.66e+6, 
+                 2.00e+6, 
+                 2.50e+6, 
+                 3.00e+6, 
+                 4.00e+6, 
+                 5.00e+6, 
+                 6.50e+6, 
+                 8.00e+6, 
+                 1.00e+7, 
+                 1.20e+7, 
+                 1.40e+7, 
+                 2.00e+7]
+
+#Define source:
+def make_source(W_Shell, C_Shell, Cells):
+    Source_List = []
+    Particle_Filter = openmc.ParticleFilter('photon')
+    Total_Mesh = openmc.UnstructuredMesh(Mesh_File, library='moab')
+    for index, bound in enumerate(bounds[:-1]):
+        Mesh_Dist = openmc.stats.MeshSpatial(Total_Mesh, strengths=strengths_list[index], volume_normalized=False)
+        Energy_Dist = openmc.stats.Uniform(a=bounds[index], b=bounds[index + 1])
+        #Source strengths given by strengths_list created by Source_Mesh_Reader
+        Source = Source_List.append(openmc.IndependentSource(space=Mesh_Dist, energy=Energy_Dist, strength=summed_strengths[index], particle='photon', domains=Cells))
+    return Source_List, Particle_Filter, Total_Mesh, Mesh_Dist
+
+# Define tallies
+def tallies(W_Shell, C_Shell, Particle_Filter, Total_Mesh):

```

Comment:
given the flexibility of the method to make cells, this may make more sense if it took a list of cells